e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -1,47 +1,66 @@
-    public static final String FROM_ODS_CODE = "GPC001";
-    public static final String TO_ODS_CODE = "B86041";
-    public static final String EHR_REQUEST = "ehrRequest";
-    public static final String GPC_ACCESS_STRUCTURED = "gpcAccessStructured";
-    public static final String GPC_STRUCTURED_FILENAME_EXTENSION = "_gpc_structured.json";
+    private static final String REQUEST_ID = "041CA2AE-3EC6-4AC9-942F-0F6621CC0BFC";
+    private static final String NHS_NUMBER = "9690937286";
+    private static final String FROM_PARTY_ID = "N82668-820670";
+    private static final String TO_PARTY_ID = "B86041-822103";
+    private static final String FROM_ASID = "200000000359";
+    private static final String TO_ASID = "918999198738";
+    private static final String FROM_ODS_CODE = "GPC001";
+    private static final String TO_ODS_CODE = "B86041";
+    private static final String EHR_REQUEST = "ehrRequest";
+    private static final String GPC_ACCESS_STRUCTURED = "gpcAccessStructured";
+    private static final String GPC_ACCESS_DOCUMENTS = "gpcAccessDocuments";
+    private static final String GPC_STRUCTURED_FILENAME_EXTENSION = "_gpc_structured.json";
+    private static final String DOCUMENT_ID = "07a6483f-732b-461e-86b6-edb665c45510";
+    private static final String GPC_ACCESS_DOCUMENT_BODY_TEMPLATE = "{\"requestId\":\"test-request-id\"," +
+        "\"conversationId\":\"%s\"," +
+        "\"documentId\":\"%s\"," +
+        "\"fromAsid\":\"200000000359\"," +
+        "\"toAsid\":\"918999198738\"}";
 
     @Test
-    public void When_extractRequestReceived_Expect_ExtractStatusAddedToDatabase() throws Exception {
+    public void When_ExtractRequestReceived_Expect_ExtractStatusAndDocumentDataAddedToDatabase() throws Exception {
         String conversationId = UUID.randomUUID().toString();
         String ehrExtractRequest = IOUtils.toString(getClass()
-            .getResourceAsStream(EHR_EXTRACT_REQUEST_TEST_FILE), Charset.defaultCharset());
+                .getResourceAsStream(EHR_EXTRACT_REQUEST_TEST_FILE), Charset.defaultCharset());
         ehrExtractRequest = ehrExtractRequest.replace("%%ConversationId%%", conversationId);
         MessageQueue.sendToMhsInboundQueue(ehrExtractRequest);
 
         var ehrExtractStatus = AwaitHelper.waitFor(() -> Mongo.findEhrExtractStatus(conversationId));
 
         assertThatInitialRecordWasCreated(conversationId, ehrExtractStatus);
 
-        var accessStructured = AwaitHelper.waitFor(
-            () -> (Document) Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_STRUCTURED));
+        Instant accessedAt = Instant.now();
+        String taskId = UUID.randomUUID().toString();
+        Mongo.addAccessDocument(conversationId, DOCUMENT_ID, taskId, accessedAt);
+
+        MessageQueue.sendToMhsTaskQueue(prepareTaskQueueMessage(conversationId));
+        assertThatAccessStructuredWasFetched(conversationId, (Document) ehrExtractStatus.get(GPC_ACCESS_STRUCTURED));
```

Comment:
test fails, this needs 'extractStatusWithUpdatedFields'  instead of original ehrExtractStatus